File: cpp/274_H-Index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        if(citations.empty()) return 0;
        int len = citations.size();
        int hash[len] = {};
        for(auto c : citations){
            if(c > len) hash[len-1]++;
            else hash[c-1]++;
        }
        int count = 0;
        for(int i = len; i > 0; --i){
            if(count + hash[i-1] >= i) return i;
            count += hash[i-1];
        }
        return 0;
    }
};

File: js/274_H-Index.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function(citations) {
    if(citations.length < 1) return 0;
    var len = citations.length;
    var hash = [];
    for(var i = 0; i < len; ++i){
        hash.push(0);
    }
    for(var j = 0; j < len; ++j){
        if(citations[j] > len) hash[len-1]++;
        else hash[citations[j]-1]++;
    }
    var count = 0;
    for(var k = len; k > 0; --k){
        if(hash[k-1] + count >= k) return k;
        count += hash[k-1];
    }
    return 0;
};